runtime/doc/api.txt

```diff
@@ -2698,6 +2698,10 @@ nvim_buf_set_extmark({buffer}, {ns_id}, {line}, {col}, {opts})
                     When a character is supplied it is used as |:syn-cchar|.
                     "hl_group" is used as highlight for the cchar if provided,
                     otherwise it defaults to |hl-Conceal|.
+                  • conceal_lines: string which should be either empty or a
+                    single character. When provided, lines in the range are
```

Comment:
Here the documentation mentions "string which should be either empty or a single character", but unlike `conceal` the API actually accepts any string without check for its content, which is a bit strange.

Reply:
Well the char part is as of yet unimplemented. The flag is a string to be future-proof towards that, while it is unimplemented it doesn't need to be checked in the API. But I guess there is no reason to already document the unimplemented behavior (although I do vaguely remember discussing that here or on matrix).

Reply:
I don't remember the outcome; there may have been concern that people already rely on the missing check and set multiple characters, which will make implementing the missing feature a breaking change.

Maybe we can add the check already even if it's not strictly necessary yet?

Reply:
I think that makes sense, done. (Though the current API check for `conceal` only checks that the first char is printable anyways, not that the string is a single character. No reason to stray from that for `conceal_lines` I guess.)